include/tvm/topi/nn/group_norm.h

```diff
@@ -56,8 +59,13 @@ inline Tensor group_norm(const Tensor& data, const Tensor& gamma, const Tensor&
       new_shape.push_back(shape[i]);
     }
   }
-  auto data_reshaped = reshape(data, new_shape);
-  // reshape gamma and beta, C -> G, C/G
+  Tensor data_reshaped;
+  if (is_float16) {
+    data_reshaped = cast(reshape(data, new_shape), DataType::Float(32));
```

Comment:
I would be good if we mention the behavior of implicit type cast in the docstring.